### PR TITLE
Add convenience extension methods for IPageApplicationModelConvention

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/RazorPagesOptionsSetup.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+﻿using System;
```
Comment (jbagga) [inline src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/RazorPagesOptionsSetup.cs]: Copyright

File: src/Microsoft.AspNetCore.Mvc.RazorPages/DependencyInjection/MvcRazorPagesMvcCoreBuilderExtensions.cs [hunk 2]
```diff
@@ -52,6 +53,11 @@ public static IMvcCoreBuilder AddRazorPages(this IMvcCoreBuilder builder)
         internal static void AddServices(IServiceCollection services)
         {
             services.TryAddEnumerable(
+                ServiceDescriptor.Transient<
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Mvc.RazorPages/DependencyInjection/MvcRazorPagesMvcCoreBuilderExtensions.cs]: Super hard to read with the lines broken up. Just put this entire bit on one line.

File: src/Microsoft.AspNetCore.Mvc.RazorPages/DependencyInjection/RazorPagesOptionsExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        }
+
+        /// <summary>
+        /// Adds a <see cref="AuthorizeFilter"/> with the specified policy to the page with the specified path.
+        /// </summary>
+        /// <param name="options">The <see cref="RazorPagesOptions"/> to configure.</param>
+        /// <param name="path">The path of the Razor Page.</param>
+        /// <param name="policy">The authorization policy.</param>
+        /// <returns>The <see cref="RazorPagesOptions"/>.</returns>
+        public static RazorPagesOptions AuthorizePage(this RazorPagesOptions options, string path, string policy)
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Mvc.RazorPages/DependencyInjection/RazorPagesOptionsExtensions.cs]: Is policy allowed to be null? Same q for other extensions that take policy.

Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.RazorPages/DependencyInjection/RazorPagesOptionsExtensions.cs]: Neither the filter that we create here, nor the `AuthorizeAttribute` perform a null check for it. I think it's designed to be null.

**Pull request closed without merging**